tools/circle_plus_gen/lib/circle_plus.py

```diff
@@ -67,6 +74,42 @@ def set_train_param(self, train_param: TrainParam):
         tparam_buff = train_param.to_buff()
         self._add_metadata(self.TINFO_META_TAG, tparam_buff)
 
+    def get_number_of_operators(self, subgraph_idx=0) -> int:
```

Comment:
### background 
To append trainable indexes into metadata, we have to know a total number of operators in subgraph.